src/compiler/checker.ts

```diff
@@ -2529,35 +2530,6 @@ module ts {
                     (<GenericType>type).target = <GenericType>type;
                     (<GenericType>type).typeArguments = type.typeParameters;
                 }
-                
-                type.declaredProperties = getNamedMembers(symbol.members);
-                type.declaredCallSignatures = emptyArray;

```

Comment:
A class shouldn't have call or construct signatures (instance side). The benefit of having these unified is that if we add the capability for classes to be callable, it will just work automatically.